src/audio/kpb.c

```diff
@@ -285,7 +289,7 @@ static int kpb_bind(struct comp_dev *dev, void *data)
 	bu = (struct ipc4_module_bind_unbind *)data;
 	buf_id = IPC4_COMP_ID(bu->extension.r.src_queue, bu->extension.r.dst_queue);
 
```

Comment:
Taking into consideration that we assigning kpb sinks here, code below should be executed only when kpb module is a source module during bind operation. E.g. you can add a check:                                                                                   `if (IPC4_MOD_ID(dev->ipc_config.id) = bu->primary.r.module_id)